android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModel.kt

```diff
@@ -80,42 +79,22 @@ constructor(
         /* TODO(View all records in this category e.g. all medical history, tasks etc) */
       }
       is ProfileEvent.OverflowMenuClick -> {
-        // TODO use navigation items from config and handle these actions dynamically
-        // https://github.com/opensrp/fhircore/issues/1371
-        when (event.menuId) {
-          R.id.individual_details ->
-            event.context.launchQuestionnaire<QuestionnaireActivity>(
-              questionnaireId = FAMILY_MEMBER_REGISTER_FORM,
-              clientIdentifier = event.patientId,
-              questionnaireType = QuestionnaireType.EDIT
-            )
-          R.id.remove_family_member ->
-            event.context.launchQuestionnaire<RemoveFamilyMemberQuestionnaireActivity>(
-              questionnaireId = REMOVE_FAMILY_FORM,
-              clientIdentifier = event.patientId,
-              intentBundle = bundleOf(Pair(NavigationArg.FAMILY_ID, event.familyId))
-            )
-          R.id.record_as_anc ->
-            event.context.launchQuestionnaire<QuestionnaireActivity>(
-              questionnaireId = ANC_ENROLLMENT_FORM,
-              clientIdentifier = event.patientId,
-              questionnaireType = QuestionnaireType.DEFAULT
-            )
-          R.id.pregnancy_outcome ->
-            event.context.launchQuestionnaire<QuestionnaireActivity>(
-              questionnaireId = PREGNANCY_OUTCOME_FORM,
-              clientIdentifier = event.patientId,
-              questionnaireType = QuestionnaireType.DEFAULT
-            )
-          R.id.record_sick_child ->
-            event.context.launchQuestionnaire<QuestionnaireActivity>(
-              questionnaireId =
-                if (event.patient.dob!!.monthsPassed() < 2) SICK_CHILD_UNDER_2M_FORM
-                else SICK_CHILD_ABOVE_2M_FORM,
-              clientIdentifier = event.patientId,
-              questionnaireType = QuestionnaireType.DEFAULT
-            )
-          else -> {}
+        event.overflowMenuItemConfig?.actions?.forEach { actionConfig ->
+          when (actionConfig.workflow) {
+            ApplicationWorkflow.LAUNCH_QUESTIONNAIRE -> {
+              var intentBundle: android.os.Bundle = android.os.Bundle.EMPTY
+              if (actionConfig.params.isNotEmpty()) {
+                intentBundle =
+                  bundleOf(Pair(actionConfig.params[0].key, actionConfig.params[0].value))
+              }
+              event.context.launchQuestionnaire<QuestionnaireActivity>(
+                questionnaireId = actionConfig.questionnaire!!.id,
```

Comment:
There is a potential `NullPointerException` here when using `actionConfig.questionnaire`

Reply:
addressed